### PR TITLE
Create `Organization` entity services

### PR DESCRIPTION
**Pull request by rashidovasadbek**
Closes: #7

### CODE REVIEW

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        CancellationToken cancellationToken = default)
+    {
+        var validationResult = organizationValidator
+            .Validate(organization,
+                options => 
+                    options.IncludeRuleSets(EntityEvent.OnCreate.ToString()));
+
+        if (!validationResult.IsValid)
+            throw new ValidationException(validationResult.Errors.ToString());
+
```
Comment (botirjonabdumannonovv) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs:48]: change "ValidationException" to FluentValidation's "ValidationException" and remove ToString().

**Review by botirjonabdumannonovv — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Mappers/OrganizationMapper.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class OrganizationMapper : Profile
+{
+    /// <summary>
+    /// Initializes the mapping configuration for Organization and Organization.
+    /// </summary>
+    public OrganizationMapper()
+    {
+        CreateMap<Organization, OrganizationDto>().ReverseMap();
+    }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Mappers/OrganizationMapper.cs:19]: Remove mapper docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/OrganizationController.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        var result = await mediator.Send(organizationGetQuery, cancellationToken);
+        return result.Any() ? Ok(result) : NoContent();
+    }
+    
+    [HttpGet("{organizationId:guid}")]
+    public async Task<IActionResult> GetOrganizationByIdAsync([FromRoute] Guid organizationId, CancellationToken cancellationToken = default)
+    {
+        var result = await mediator.Send(new OrganizationGetByIdQuery{OrganizationId = organizationId}, cancellationToken);
+        return result is not null ? Ok(result) : NoContent();
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/OrganizationController.cs:23]: Change to `NotFound` result when entity couldn't be found

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/OrganizationController.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        var result = await mediator.Send(command, cancellationToken);
+        return Ok(result);
+    }
+    
+    [HttpDelete("{organizationId:guid}")]
+    public async ValueTask<IActionResult> DeleteOrganizationByIdAsync([FromRoute] Guid organizationId, CancellationToken cancellationToken = default)
+    {
+       var result =  await mediator.Send(new DeleteOrganizationByIdCommand{ OrganizationId = organizationId}, cancellationToken);
+        return  result ? Ok() : BadRequest();
+    }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/OrganizationController.cs:45]: Remove `default` value for `CancellationToken`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/OrganizationController.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+    [HttpPut]
+    public async Task<IActionResult> UpdateOrganizationAsync([FromBody] UpdateOrganizationCommand command, CancellationToken cancellationToken = default)
+    {
+        var result = await mediator.Send(command, cancellationToken);
+        return Ok(result);
+    }
+    
+    [HttpDelete("{organizationId:guid}")]
+    public async ValueTask<IActionResult> DeleteOrganizationByIdAsync([FromRoute] Guid organizationId, CancellationToken cancellationToken = default)
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/OrganizationController.cs:41]: Remove `Async` postfix from action names

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Organizations/Models/OrganizationDto.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    
+    /// <summary>
+    ///  Gets or sets the  description of the Organization.
+    /// </summary>
+    public string Description { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets a collection of Clients.
+    /// </summary>
+    public Client Client { get; set; } = default!;
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Organizations/Models/OrganizationDto.cs:34]: Change to `ClientDto` type

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Organizations/Queries/OrganizationGetQuery.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿using Feedback.Analyzer.Application.Organizations.Models;
+using Feedback.Analyzer.Domain.Common.Queries;
+
+namespace Feedback.Analyzer.Application.Organizations.Queries;
+
+public class OrganizationGetQuery : IQuery<ICollection<OrganizationDto>>
+{
+    public OrganizationFilter OrganizationFilter { get; init; } = default!;
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Organizations/Queries/OrganizationGetQuery.cs:9]: Add docs, remove `init`

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/QueryHandlers/OrganizationGetQueryHandler.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+public class OrganizationGetQueryHandler(IOrganizationService organizationService, IMapper mapper) : IQueryHandler<OrganizationGetQuery, ICollection<OrganizationDto>>
+{
+    public async Task<ICollection<OrganizationDto>> Handle(OrganizationGetQuery organizationGetQuery, CancellationToken cancellationToken)
+    {
+        var matchedOrganizations =  await organizationService.Get(organizationGetQuery.OrganizationFilter, new QueryOptions() { AsNoTracking = true }).ToListAsync(cancellationToken);
+        
+        return mapper.Map<ICollection<OrganizationDto>>(matchedOrganizations);
+    }
+
+
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/QueryHandlers/OrganizationGetQueryHandler.cs:23]: Remove extra lines

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Domain.Enums;
+using Feedback.Analyzer.Infrastructure.Organizations.Validators;
+using Feedback.Analyzer.Persistence.Extensions;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+using FluentValidation;
+
+namespace Feedback.Analyzer.Infrastructure.Organizations.Services;
+
+public class OrganizationService(
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Organizations/Services/OrganizationService.cs:15]: Add type docs

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — APPROVED** [review]

**Review by isxoqovxasanboy — APPROVED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by Habiba0206 — APPROVED** [review]

**Pull request merged**